src/OpenLoco/Windows/Vehicle.cpp

```diff
@@ -2265,13 +2265,129 @@ namespace OpenLoco::Ui::Vehicle
+        {
+            return Vehicles::OrderRingView(head->orderTableOffset);
+        }
+
+        static std::pair<Map::map_pos3, const char*> sub_470B76(const Vehicles::Order& order, uint8_t orderNum)
         {
+            static loco_global<char[512], 0x0112CC04> _stringFormatBuffer;
+
             registers regs{};
-            regs.esi = reinterpret_cast<uint32_t>(head);
+            regs.ebp = order.getOffset();
+            regs.ebx = static_cast<uint8_t>(order.getType());
+            regs.ecx = orderNum;
             call(0x00470824, regs);
+            Map::map_pos3 res = { regs.ax, regs.cx, regs.dx };
+            return std::make_pair(res, _stringFormatBuffer.get());
+        }
+
+        // 0x00470824
+        static void sub_470824(Vehicles::VehicleHead* head)
+        {
+            Input::setMapSelectionFlags(Input::MapSelectionFlags::unk_04);
+            Gfx::invalidateScreen();
+            for (auto& unk : _F2494A)
+            {
+                unk.orderOffset = static_cast<uint32_t>(-1);
+            }
+            auto orders = getOrderTable(head);
+            uint8_t i = 0;
+            for (auto& order : orders)
+            {
+                if (!order.hasFlag(Vehicles::OrderFlags::HasNumber))
+                {
+                    continue;
+                }
+
+                _F2494A[i].orderOffset = order.getOffset();
+
+                auto lineNumber = 0;
+                auto* stationOrder = order.as<Vehicles::OrderStation>();
+                auto* waypointOrder = order.as<Vehicles::OrderRouteWaypoint>();
+                if (stationOrder != nullptr)
+                {
+                    lineNumber++; // station labels start on line 0 so add 1 for the number
+
+                    const auto stationId = stationOrder->getStation();
+                    for (auto innerOrder = orders.begin(); innerOrder->getOffset() != order.getOffset(); ++innerOrder)
+                    {
+                        auto* innerStationOrder = innerOrder->as<Vehicles::OrderStation>();
+                        if (innerStationOrder == nullptr)
+                        {
+                            continue;
+                        }
+                        if (stationId == innerStationOrder->getStation())
+                        {
+                            lineNumber++;
+                        }
+                    }
+                }
+                else if (waypointOrder != nullptr)
+                {
+                    const auto rawOrder = order.getRaw();
+                    for (auto innerOrder = orders.begin(); innerOrder->getOffset() != order.getOffset(); ++innerOrder)
+                    {
+                        if (rawOrder == innerOrder->getRaw())
+                        {
+                            lineNumber++;
+                        }
+                    }
+                }
+
+                _F2494A[i].lineNumber = lineNumber & 0xFF;
```

Comment:
Is the & 0xFF needed when assigning to an uint8?

Reply:
no but it was to draw attention to this stupidity of counting in a uint16 then saving as a uint8